modules/git/repo_commit.go

```diff
@@ -203,35 +203,55 @@ func (repo *Repository) commitsByRange(id SHA1, page int) (*list.List, error) {
 
 func (repo *Repository) searchCommits(id SHA1, opts SearchCommitsOptions) (*list.List, error) {
 	cmd := NewCommand("log", id.String(), "-100", "-i", prettyLogFormat)
-	if len(opts.Keywords) > 0 {
-		for _, v := range opts.Keywords {
-			cmd.AddArguments("--grep=" + v)
-		}
-	}
+	args := []string{"log", "-i", prettyLogFormat}
 	if len(opts.Authors) > 0 {
 		for _, v := range opts.Authors {
 			cmd.AddArguments("--author=" + v)
+			args = append(args, "--author="+v)
 		}
 	}
 	if len(opts.Committers) > 0 {
 		for _, v := range opts.Committers {
 			cmd.AddArguments("--committer=" + v)
+			args = append(args, "--committer="+v)
 		}
 	}
 	if len(opts.After) > 0 {
 		cmd.AddArguments("--after=" + opts.After)
+		args = append(args, "--after="+opts.After)
 	}
 	if len(opts.Before) > 0 {
 		cmd.AddArguments("--before=" + opts.Before)
+		args = append(args, "--before="+opts.Before)
 	}
 	if opts.All {
 		cmd.AddArguments("--all")
+		args = append(args, "--all")
+	}
+	if len(opts.Keywords) > 0 {
+		for _, v := range opts.Keywords {
+			cmd.AddArguments("--grep=" + v)
```

Comment:
This should be moved to the `for _, v := range opts.Keywords {` at the bottom of the function.
I think the outcomes will be no different.